public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -57,6 +57,9 @@ export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatas
       if (!(panel.parent instanceof LibraryVizPanel)) {
         throw new Error('Could not find SceneQueryRunner for panel');
       } else {
+        if (!panel.parent.isActive) {
+          panel.parent.activate();
+        }
```

Comment:
need to call the deactivation function returned by activate (in the deactivation function for this behavior)